### PR TITLE
Added func test for case creation idempotency

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-825

### Change description ###
Added func test for case creation idempotency

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 1]
```diff
@@ -92,6 +94,44 @@ public void should_create_case_from_valid_exception_record() throws Exception {
...
+        assertThat(createdCase.getCaseTypeId()).isEqualTo(BULK_SCANNED_CASE_TYPE);
+        assertThat(createdCase.getData().get("firstName")).isEqualTo("value1");
+        assertThat(createdCase.getData().get("lastName")).isEqualTo("value2");
+        assertThat(createdCase.getData().get("email")).isEqualTo("hello@test.com");
+
+        assertThat(createdCase.getData().get("bulkScanCaseReference")).isNotNull();
+        String bulkScanCaseReference = (String)createdCase.getData().get("bulkScanCaseReference");
+
+        assertThat(ccdApi.getCaseRefsByBulkScanCaseReference(bulkScanCaseReference, "bulkscan").size())
+            .isEqualTo(1);
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: is it not enough to just assert reference is the one which is in exception record?

Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Have added checks

File: config/owasp/suppressions.xml [hunk 2]
```diff
@@ -119,9 +119,10 @@
     <packageUrl regex="true">^pkg:maven/com\.fasterxml\.jackson\.core/jackson\-databind@.*$</packageUrl>
     <cve>CVE-2019-16942</cve>
     <cve>CVE-2019-16943</cve>
+    <cve>CVE-2019-17531</cve>
   </suppress>
 
-  <suppress>
+  <suppress until="2019-10-31">
```
Comment (doncem) [inline config/owasp/suppressions.xml]: Why PRs always include some of the work which is already done elsewhere?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 3]
```diff
@@ -92,6 +94,46 @@ public void should_create_case_from_valid_exception_record() throws Exception {
...
+        assertThat(createdCase.getData().get("firstName")).isEqualTo("value1");
+        assertThat(createdCase.getData().get("lastName")).isEqualTo("value2");
+        assertThat(createdCase.getData().get("email")).isEqualTo("hello@test.com");
+
+        assertThat(createdCase.getData().get("bulkScanCaseReference")).isNotNull();
+        String bulkScanCaseReference = (String)createdCase.getData().get("bulkScanCaseReference");
+
+        List<Long> caseIds = ccdApi.getCaseRefsByBulkScanCaseReference(bulkScanCaseReference, "bulkscan");
+        assertThat(caseIds.size()).isEqualTo(1);
+        assertThat(caseIds.get(0)).isEqualTo(createdCase.getId());
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: so how about just a one liner here?

```java
assertThat(createdCase.getData().get("bulkScanCaseReference")).equals(exceptionRecord.getId());
```

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 4]
```diff
@@ -92,6 +94,46 @@ public void should_create_case_from_valid_exception_record() throws Exception {
...
+
+        Thread.sleep(2000L);
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: is sleeping needed?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java (hunk 3 above)
Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Updated this piece of code.

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java (hunk 4 above)
Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Updated - used await... instead. This is how it is described in the task - wait for 2 sec until ES is refreshed

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 5]
```diff
@@ -92,6 +94,46 @@ public void should_create_case_from_valid_exception_record() throws Exception {
...
+
+        await("Case is ingested")
+            .atMost(2, TimeUnit.SECONDS)
+            .pollDelay(1, TimeUnit.SECONDS)
+            .until(() -> caseIngested(bulkScanCaseReference));
+
+        // when
+        // create case callback endpoint invoked second time
+        AboutToStartOrSubmitCallbackResponse callbackResponse2 = invokeCallbackEndpoint(exceptionRecord);
+        String caseCcdId2 = getCaseCcdId(callbackResponse2);
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: can we do all setup in one go?
ie:
1. send request for the 1st time,
2. send request for the 2nd time,
3. make assertions

Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: The idea here is to replicate the test above (should_create_case_from_valid_exception_record) and add the check for idempotency. The test above will be deleted once this test starts passing.

**Pull request merged**